lib/snapshot_logger/snapshot_logger.ex

```diff
@@ -66,9 +66,23 @@ defmodule SnapshotLogger.SnapshotLogger do
         snapshot_data = make_snapshot(module, state)
         Map.put(acc, module_name(module), snapshot_data)
       end)
-      |> Jason.encode_to_iodata!()
 
-    Logger.info(snapshot)
+    best_f_measure_snapshot =
+      snapshot
+      |> Enum.map(fn {key, value} ->
+        new_value =
+          value
+          |> Enum.map(&Map.take(&1, [:value, :f_measure]))
+          |> Enum.sort_by(& &1.f_measure, :desc)
+          |> hd()
+
+        {key, new_value}
+      end)
+      |> Enum.into(%{})
```

Comment:
suggestion (non-blocking): There's actually a `Enum.into/3` which takes a transformation function as the 3rd argument, so you don't actually need the separate `Enum.map` line above.

Reply:
oh, nice!